lib/queue.rb

```diff
@@ -1,31 +1,58 @@
 class Queue
 
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(220)
+    @front = @back = -1
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if @front == -1 && @back == -1
+      @front = 0
+      @back = 0
+    elsif @front == @back
+      raise StandardError.new "Cannot add element. Queue is full."
+    end
+
+    @store[@back] = element
+    @back = (@back + 1) % @store.length
   end
 
   def dequeue
-    raise NotImplementedError, "Not yet implemented"
+    element = @store[@front]
+
+    @store[@front] = nil
+    @front = (@front + 1) % @store.length
+
+    return element
```

Comment:
You should also check that the queue is empty now.

```suggestion
    element = @store[@front]

    @store[@front] = nil
    @front = (@front + 1) % @store.length
    @front = @back = -1 if @front == @back

    return element
```